python/ray/tune/trial_runner.py

```diff
@@ -322,14 +323,26 @@ def __init__(self,
 
         self._callbacks = CallbackList(callbacks or [])
 
-        self._callbacks.setup()
-
         if checkpoint_period is None:
             checkpoint_period = os.getenv("TUNE_GLOBAL_CHECKPOINT_S", "auto")
 
         self._checkpoint_period = checkpoint_period
         self._checkpoint_manager = self._create_checkpoint_manager()
 
+    def setup_callbacks(self, experiment: Optional[Experiment]) -> None:
+        """Calls ``setup`` method in callbacks.
+
+        Args:
+            experiment (Experiment): Experiment which ``public.spec`` is
+                passed to callback ``setup`` methods.
+        """
+        spec = experiment.public_spec if experiment else {}
+        self._callbacks.setup(**spec)
+
+    def end_experiment_callbacks(self) -> None:
+        """Calls ``on_experiment_end`` method in callbacks."""
+        self._callbacks.on_experiment_end(trials=self._trials)
```

Comment:
Can this be included in `cleanup_trials`?

Reply:
It can, though `cleanup_trials` wasn't explicit enough for me. The name of it implies that it will only handle trials.

Reply:
Rename it to `cleanup`?

Reply:
Ok, done